src/inmanta/db/versions/v202511180.py

```diff
@@ -0,0 +1,36 @@
+"""
+Copyright 2025 Inmanta
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Contact: code@inmanta.com
+"""
+
+from asyncpg import Connection
+
+
+async def update(connection: Connection) -> None:
+    """
+    Make discovery_resource_id a mandatory column.
+    """
+    schema = """
+    -- Drop all discovered resources that were creating using the previous major
+    -- release of the orchestrator.
+    DELETE FROM public.discoveredresource
+    WHERE discovery_resource_id IS NULL;
```

Comment:
The ticket stated "Set null values to some sort of legacy / unknown resource id". I don't have a very strong opinion, but that does seem safest to me.